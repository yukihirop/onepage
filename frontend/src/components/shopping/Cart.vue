<template lang="pug">
  .cart
    h1.title Your Cart
    p(v-show='!products.length')
      i Your cart is empty!
      router-link(to='/') Go shopping

    table.table.is-striped(v-show='products.length')
      thead
        tr
          td Name
          td Price
          td Quantity
      tbody
        tr(v-for='product in products')
          td {{ product.name }}
          td {{ product.price }}
          td {{ product.quantity }}
        tr
          td
            b Total:
          td
          td
            b ${{ total }}
    p
      button.button.is-primary(v-show='products.length', @click='checkout') Checkout
</template>

<script>
import { mapGetters } from 'vuex'
const namespace = 'shoppingModule'

export default {
  computed: {
    ...mapGetters(namespace, {
      products: 'cartProducts'
    }),
    total () {
      return this.products.reduce((total, product) => {
        return total + product.price * product.quantity
      }, 0)
    }
  },
  methods: {
    checkout () {
      alert('Play us $' + this.total)
    }
  }
}
</script>