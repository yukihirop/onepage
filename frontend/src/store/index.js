import Vue from 'vue'
import Vuex from 'vuex'
import shoppingModule from './modules/shopping'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

export default new Vuex.Store({
  strict: debug,
  modules: {
    shoppingModule
  }
})