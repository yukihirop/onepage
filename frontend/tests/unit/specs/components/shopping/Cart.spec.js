import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import Cart from '@/components/shopping/Cart.vue'
import store from '@/store'
import factoryCart from '@specs/factories/components/shopping/Cart'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Cart.vue', () => {
  it('初期状態のレンダリングのテスト', () => {
    const wrapper = shallow(Cart, { store, localVue })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })

  it('storeを用意してのレンダリングのテスト', () => {
    const mockedStore = {
      modules: {
        shoppingModule: {
          namespaced: true,
          getters: {
            cartProducts: jest.fn().mockReturnValue(factoryCart.cartProducts)
          },
          methods: {
            checkout (state) {
              jest.fn()
            }
          }
        }
      }
    }
    const store = new Vuex.Store(mockedStore)
    const wrapper = shallow(Cart, { store })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
