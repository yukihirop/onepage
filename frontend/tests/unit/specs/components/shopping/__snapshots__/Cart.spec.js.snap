// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart.vue storeを用意してのレンダリングのテスト 1`] = `
<div class="cart">
    <h1 class="title">Your Cart</h1>
    <p style="display: none;">
        <i>Your cart is empty!</i>
        <router-link to="/">Go shopping</router-link>
    </p>
    <table class="table is-striped">
        <thead>
            <tr>
                <td>Name</td>
                <td>Price</td>
                <td>Quantity</td>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>a</td>
                <td>1</td>
                <td>10</td>
            </tr>
            <tr>
                <td>b</td>
                <td>2</td>
                <td>20</td>
            </tr>
            <tr>
                <td>c</td>
                <td>3</td>
                <td>30</td>
            </tr>
            <tr>
                <td>
                    <b>Total:</b>
                </td>
                <td></td>
                <td>
                    <b>$140</b>
                </td>
            </tr>
        </tbody>
    </table>
    <p>
        <button class="button is-primary">Checkout</button>
    </p>
</div>
`;

exports[`Cart.vue 初期状態のレンダリングのテスト 1`] = `
<div class="cart">
    <h1 class="title">Your Cart</h1>
    <p>
        <i>Your cart is empty!</i>
        <router-link to="/">Go shopping</router-link>
    </p>
    <table class="table is-striped" style="display: none;">
        <thead>
            <tr>
                <td>Name</td>
                <td>Price</td>
                <td>Quantity</td>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td>
                    <b>Total:</b>
                </td>
                <td></td>
                <td>
                    <b>$0</b>
                </td>
            </tr>
        </tbody>
    </table>
    <p>
        <button class="button is-primary" style="display: none;">Checkout</button>
    </p>
</div>
`;
